iotfunctions/anomaly.py

```diff
@@ -507,10 +509,18 @@ def prepare_data(self, dfEntity):
     def execute(self, df):
 
         df_copy = df.copy()
-        entities = np.unique(df_copy.index.levels[0])
-        logger.debug(str(entities))
+        unique_entities = np.unique(df_copy.index.levels[0])
+        entities = []
+        if self.entity_list:
+            for entity in self.entity_list:
+                if entity in unique_entities:
+                    entities.append(entity)
+        else:
+            entities = unique_entities
 
-        df_copy[self.output_item] = 0
+        logger.debug('Entities to be processed {}'.format(str(entities)))
+
+        df_copy[self.output_item] = np.nan
```

Comment:
Please check whether the output_item column has already been created. Otherwise you overwrite scores from previous invocations of KMeans with different filter settings defeating the purpose of the filters to process subsets of entities with entity group specific settings.
```
if self.output_item is in df_copy.columns:
        df_copy[self.output_item] = np.nan
```

Reply:
This one isn't clear to me. If their were previous invocations of KMeans with different filter settings i.e. filter settings were changed then KPI is recalculated. We can talk about this more.

Reply:
As per slack we can cheat and force more than one function to write into the same output column - which doesn't make sense in general but would be very valuable in our case.
So let's discuss whether the pipeline could support it and what changes in the UI would be required to do it. In that light I'd advocate to check for existing columns

Reply:
I need to discuss this with you.

Reply:
Now, I understood the above change